### PR TITLE
updating GenomicsDB integration to match the changes in the importer

### PR DESCRIPTION
**Pull request by lbergelson**
* added a reference parameter to FeatureData source and FeatureManager methods
  * genomicsDB requires a reference, previously this was being passed
  through by hardcoding it in the required json files
  * json files are now autogenerated by the importer tool, but the
  reference wasn't being handled correctly

* updated the various walkers to pass the reference through if available

* gendb:// paths now point to the workspace directory instead of a
directory of jsons

* removed the ability to specify array, vidmap.json, and
callset.json paths in the importer tool since we now rely on the
structure and naming of the files when loading
moved some constants to GenomicsDBConstants

* updated GenomicsDBIntegration tests to use the new importer instead of a
prepackaged and very brittle set of json files

fixed a bug in GenomicsDBImporterIntegrationTests that made both tests
write to the same workspace

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java [hunk 1]
```diff
@@ -57,7 +59,8 @@ private void initializeDrivingFeatures() {
             drivingFeatures = new FeatureDataSource<>(new FeatureInput<>(drivingFile.getAbsolutePath()), FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES, null, cloudPrefetchBuffer, cloudIndexPrefetchBuffer);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java]: Forgot to pass in the reference here

File: src/main/java/org/broadinstitute/hellbender/engine/MultiVariantWalker.java [hunk 2]
```diff
@@ -61,7 +62,8 @@ protected void initializeDrivingVariants() {
 
                     //Add the driving datasource to the feature manager too so that it can be queried. Setting lookahead to 0 to avoid caching.
                     //Note: we are disabling lookahead here because of windowed queries that need to "look behind" as well.
-                    features.addToFeatureSources(0, featureInput, VariantContext.class, cloudPrefetchBuffer, cloudIndexPrefetchBuffer);
+                    features.addToFeatureSources(0, featureInput, VariantContext.class, cloudPrefetchBuffer, cloudIndexPrefetchBuffer,
+                                                 referenceArguments.getReferencePath());
                 }
         );
         drivingVariants = new MultiVariantDataSource(drivingVariantsFeatureInputs, VariantWalkerBase.FEATURE_CACHE_LOOKAHEAD, cloudPrefetchBuffer, cloudIndexPrefetchBuffer);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/MultiVariantWalker.java]: We currently have no way to pass in the reference to a `MultiVariantDataSource`

File: src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java [hunk 3]
```diff
@@ -38,56 +38,85 @@ public void testGenomicsDBInClassPath(){
...
-            final ArgumentsBuilder args = baseArgs
-                    .addOutput(output);
-            runCommandLine(args);
-        }, new File(WORKSPACE), LOADER_JSON);
+        final ArgumentsBuilder args = baseArgs
+                .addOutput(output);
+        runCommandLine(args);
 
         final Iterable<VariantContext> actualVcs = new FeatureDataSource<>(output);
         final Iterable<VariantContext> expectedVcs = new FeatureDataSource<>(expected);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java]: Close your data sources using try-with-resources

File: src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java [hunk 4]
```diff
@@ -38,56 +38,85 @@ public void testGenomicsDBInClassPath(){
...
 
         BaseTest.assertCondition(actualVcs, expectedVcs, (a, e) -> VariantContextTestUtils.assertVariantContextsAreEqual(a,e, Collections.emptyList()));
     }
 
+    private static String createTempGenomicsDB(final List<File> gvcfs, final Locatable interval) {
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java]: Do we have any test cases that pass in a List of size > 1 here?

**Comment by droazen** [comment]
Review complete -- mostly minor comments, nothing critical. Given the need to get this merged today, I'll merge this now if you promise to address the comments in a separate branch :)

**Comment by lbergelson** [comment]
creating #2628 to remind us about this

File: src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java (hunk 1 above)
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java]: done

File: src/main/java/org/broadinstitute/hellbender/engine/MultiVariantWalker.java (hunk 2 above)
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/MultiVariantWalker.java]: added another parameter

File: src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java (hunk 3 above)
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java]: good catch, done

File: src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java (hunk 4 above)
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/engine/GenomicsDBIntegrationTest.java]: No, made it a scalar instead

**Pull request merged**